docs/en/ingest-management/fleet-agent-proxy-support.asciidoc

```diff
@@ -0,0 +1,252 @@
+* Retrieve agent policies from {es} (only needed for agents running {fleet-server})
+* Monitor services (not shown)
+
+image::images/agent-proxy-server.png[Image showing connections between {agent}, {fleet-server}, and {es}]
+
+You can configure {agent} proxy settings by specifying:
+
+* <<host-proxy-env-vars>>
+* <<proxy-settings-in-agent-policy>>
+* <<cli-proxy-settings>>
+
+[discrete]
+[[host-proxy-env-vars]]
+=== Host environment variables
+
+Set environment variables on the host to configure default proxy settings.
+The {agent} uses host environment settings by default if no proxy settings are
+specified elsewhere. You can override host proxy settings later when you
+configure the {agent} and {fleet} settings. The following environment variables
+are available on the host:
+
+|===
+|Variable |Description
+
+|`HTTP_PROXY`
+|URL of the proxy server for HTTP traffic. 
+
+|`HTTPS_PROXY`
+|URL of the proxy server for HTTPS traffic.
+
+|`NO_PROXY`
+|IP addresses or domain names that should not use the proxy. Supports patterns.
+|===
+
+The proxy URL can be a complete URL or `host[:port]`, in which case the `http`
+scheme is assumed. An error is returned if the value is a different form.
+
+Keep in mind that proxy environment settings may affect other components on the
+host that send HTTP requests.
+
+//TODO: Add examples of how/where to set env vars on various platforms.
+
+[discrete]
+[[proxy-settings-in-agent-policy]]
+=== Proxy settings in the {agent} policy
+
+Proxy settings in the {agent} policy override proxy settings specified by
+environment variables. The following proxy settings are valid in the agent
+policy:
+
+|===
+|Setting | Description
+
+|`proxy_url`
+| (string) URL of the proxy server. If set, the configured URL is used as a
+proxy for all connection attempts by the component. The value may be either a
+complete URL or a `host[:port]`, in which case the `http` scheme is assumed. If
+a value is not specified through the configuration, then proxy environment
+variables are used. The URL accepts optional `username` and `password` settings
+for authenticating with the proxy. For example:
+`http://<username>:<password>@<proxy host>/`.
+
+|`proxy_headers`
+| (string) Additional headers to send to the proxy during CONNECT requests. You
+can use this setting to pass keys/tokens required for authenticating with the
+proxy.
+
+|`proxy_disable`
+| (boolean) If set to `true`, all proxy settings, including the `HTTP_PROXY` and
+`HTTPS_PROXY` environment variables, are ignored.
+
+|===
+
+Proxy settings are valid in the {agent} policy under the following sections:
+
+* `outputs` - set the proxy server for communicating with {es}.
+
+* `fleet` - set the proxy server for retrieving agent policies from
+{fleet}.
+
+* `agent.download` - set the proxy server for retrieving artifacts or binary
+upgrades from `artifacts.elastic.co`.
+
+//TODO: Clarify where to add these settings ^^. For Fleet-managed agents, it
```

Comment:
@michel-laterman See my comment here. Can you help me with some examples that show these settings for Fleet-managed and self-managed agents (including YAML configs, which file they're in, screen captures, if relevant). I feel I am not connecting the dots completely here. I suspect the flow of info on this page isn't quite right either but it's hard to know without having all the technical details.